packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -327,14 +327,18 @@ async function getParameterJson(ctx: SolutionContext) {
 
   let parameterJson;
   if (createNewParameterFile) {
-    parameterJson = await getExpandedParameter(ctx, parameterTemplateFilePath, false); // do not expand secrets to avoid saving secrets to parameter file
     await fs.ensureDir(parameterFolderPath);
-    await fs.writeFile(parameterFilePath, JSON.stringify(parameterJson, undefined, 2));
+    if (isMultiEnvEnabled()) {
+      await fs.copyFile(parameterTemplateFilePath, parameterFilePath);
+    } else {
+      parameterJson = await getExpandedParameter(ctx, parameterTemplateFilePath, false); // do not expand secrets to avoid saving secrets to parameter file
+      await fs.writeFile(parameterFilePath, JSON.stringify(parameterJson, undefined, 2));
+    }
   }
 
   parameterJson = await getExpandedParameter(ctx, parameterFilePath, true); // only expand secrets in memory
 
-  return parameterJson;
+  return [parameterFilePath, parameterJson];
```

Comment:
seems the only purpose of returning file path is print some logs, can we print the logs in solution plugin instead of core? the return value seems a little mismatch with the function name

Reply:
Ok, let me remove the log in core part.